playbooks/common/openshift-cluster/update_repos_and_packages.yml

```diff
@@ -5,8 +5,10 @@
   roles:
   - role: rhel_subscribe
     when: deployment_type in ["enterprise", "atomic-enterprise", "openshift-enterprise"] and
-          ansible_distribution == "RedHat" and
-          lookup('oo_option', 'rhel_skip_subscription') | default(rhsub_skip, True) |
-            default('no', True) | lower in ['no', 'false']
+          ansible_distribution == "RedHat" and not ( openshift_rhel_skip_subscription |
+          default(lookup('oo_option', 'rhel_skip_subscription')) | default(rhsub_skip) |
+          default(False) | bool )
```

Comment:
Also, if using the set fact to merge the oo_option lookup vars under the openshift_* vars, you can push the conditional logic down into the role itself instead of conditionalizing the role inclusion in the playbook (since the when on the role simply applies the when conditional to all tasks in the role anyway).